Models/user.rb

```diff
@@ -7,16 +7,89 @@ class User
   @password
   @mattress
   @account
-  @pockets = []
-  @goals = []
-  def initialize(mysql_obj) 
+  @mysql_obj
+
+  attr_accessor :id, :password, :mattress, :pockets, :goals, :first_name, :last_name, :email, :account
+  def initialize(mysql_obj, id)
+    @pockets = []
+    @goals = []
+    @id = id
+    @mysql_obj = mysql_obj
+    define_attributes
+    @mattress = Mattress.new(@mysql_obj, @id)
+    @account = Account.new(@mysql_obj, @id)
+    define_pockets
+    define_goals
   end
 
   def available
-    @account.balance
+    @account.available
+  end
+
+  def balance_total
+    sum_total = @account.available + @mattress.save_money
+    @pockets.each do |pocket|
+      sum_total += pocket.balance
+    end
+    @goals.each do |goal|
+      sum_total += goal.current_amount
+    end
+    sum_total
+  end
+
+  def list_transactions(n_transactions)
+    count = 0
+    transactions = @mysql_obj.query("(SELECT ext.type,ext.amount,ext.transaction_date FROM `external_trasactions` AS ext INNER JOIN users ON ext.user_id = users.id WHERE users.id = '#{@id}' ORDER BY ext.transaction_date DESC) UNION (SELECT CONCAT(IF(user_id_origin = '#{@id}','send','reception')) as type, tx.amount,tx.transaction_date FROM `transfers` AS tx INNER JOIN users ON tx.user_id_destination = users.id OR tx.user_id_origin = users.id WHERE users.id = '#{@id}' ORDER BY tx.transaction_date DESC) ORDER BY `transaction_date` DESC")
+    transactions.each do |row|
+      puts row
+      count += 1
+      break if count >= n_transactions
+    end
+    gets
```

Comment:
Este gets que funcion cumple?

Reply:
```suggestion
    gets
```
@FernandoMorantes 
ese gets se me paso eso solo para que se quede esperando un enter